core/shared/src/main/scala/fs2/interop/flow/StreamSubscriber.scala

```diff
@@ -267,39 +277,51 @@ private[flow] final class StreamSubscriber[F[_], A] private (
 
         case Idle(s) =>
           WaitingOnUpstream(cb, s) -> run {
-            s.request(chunkSize.toLong)
             // We do the updates here,
             // to ensure they happen after we have secured the state.
             inOnNextLoop = false
-            index = 0
+            buffer = null
```

Comment:
In theory redundant, but just in case, we make sure we start in the initial state.
The `index` however felt wrong setting it to `0` here since it doesn't matter, the `index` is only used when we are in the `onNext` loop. Thus, I made sure that the use of that value only happens in interactions directly related to that loop.